nodes/basic-pow/src/service.rs

```diff
@@ -29,93 +35,104 @@ pub fn build_inherent_data_providers() -> Result<InherentDataProviders, ServiceE
 	Ok(providers)
 }
 
-/// Starts a `ServiceBuilder` for a full service.
-///
-/// Use this macro if you don't actually need the full service, but just the builder in order to
-/// be able to perform chain operations.
-macro_rules! new_full_start {
-	($config:expr) => {{
-		let mut import_setup: Option<_> = None;
-		let inherent_data_providers = crate::service::build_inherent_data_providers()?;
-
-		let builder = sc_service::ServiceBuilder::new_full::<
-			runtime::opaque::Block,
-			runtime::RuntimeApi,
-			crate::service::Executor,
-		>($config)?
-		.with_select_chain(|_config, backend| Ok(sc_consensus::LongestChain::new(backend.clone())))?
-		.with_transaction_pool(|builder| {
-			let pool_api = sc_transaction_pool::FullChainApi::new(builder.client().clone());
-			Ok(sc_transaction_pool::BasicPool::new(
-				builder.config().transaction_pool.clone(),
-				std::sync::Arc::new(pool_api),
-				builder.prometheus_registry(),
-			))
-		})?
-		.with_import_queue(
-			|_config, client, select_chain, _transaction_pool, spawn_task_handle, registry| {
-				let pow_block_import = sc_consensus_pow::PowBlockImport::new(
-					client.clone(),
-					client.clone(),
-					sha3pow::MinimalSha3Algorithm,
-					0, // check inherents starting at block 0
-					select_chain,
-					inherent_data_providers.clone(),
-				);
-
-				let import_queue = sc_consensus_pow::import_queue(
-					Box::new(pow_block_import.clone()),
-					None,
-					None,
-					sha3pow::MinimalSha3Algorithm,
-					inherent_data_providers.clone(),
-					spawn_task_handle,
-					registry,
-				)?;
-
-				import_setup = Some(pow_block_import);
-
-				Ok(import_queue)
-			},
-		)?;
-
-		(builder, import_setup, inherent_data_providers)
-		}};
+/// Returns most parts of a service. Not enough to run a full chain,
+/// But enough to perform chain operations like purge-chain
+pub fn new_full_params(config: Configuration) -> Result<(
+	sc_service::ServiceParams<
+		Block, FullClient,
+		BasicQueue<Block, TransactionFor<FullClient, Block>>,
+		sc_transaction_pool::FullPool<Block, FullClient>,
+		(), FullBackend,
+	>,
+	FullSelectChain,
+	sp_inherents::InherentDataProviders,
+	sc_consensus_pow::PowBlockImport<Block>,
```

Comment:
@expenses Can you help me here too? I'm trying to return my `pow_block_import` from the `new_full_params` function so it can be used in `new_full`. I just can't figure out what type to put here.